service/storage_service.cc

```diff
@@ -2870,7 +2870,13 @@ future<> storage_service::restore_replica_count(inet_address endpoint, inet_addr
     }
   return seastar::async([this, endpoint, notify_endpoint] {
     auto tmptr = get_token_metadata_ptr();
-    auto streamer = make_lw_shared<dht::range_streamer>(_db, tmptr, _abort_source, get_broadcast_address(), "Restore_replica_count", streaming::stream_reason::removenode);
+    abort_source as;
+    auto sub = _abort_source.subscribe([&as] {
+        if (!as.abort_requested()) {
```

Comment:
cosmetic nit: not really needed to check for `abort_requested()` prior to `request_abort()`.

Reply:
I wanted to avoid exception, there are multiple places we can request the abort. Nothing guarantees the as is not aborted when _abort_source calls the lambda.

Reply:
Calling request_abort() on an already aborted abort_source is a no-op, it doesn't throw.

Reply:
Are you sure?

```
std::optional<subscription_list_type> _subscriptions

114     void request_abort() { 
115         _subscriptions->clear_and_dispose([] (subscription* s) { s->on_abort(); });
116         _subscriptions = { };
117     }
```

It is not a no-op but a segmentation fault.

Reply:
You're right, so the if is needed.

We should fix request_abort(), it's a usability flaw and a safety risk.

Reply:
Yes, we can fix reqeeust_abort(), but I do not want to make this patch depend on it, we need to backport this to stable branch.